src/DiagramGenerator/Diagram/Board.php

```diff
@@ -315,25 +321,26 @@ protected function getPieceImagePath(Piece $piece)
     /**
      * Returns board background image path
      *
-     * @return string
+     * @return \Imagick
+     *
+     * @throws \ImagickException
      */
     protected function getBackgroundTexture()
     {
-        $boardCachedPath = $this->cacheDir . '/board/' . $this->getBoardTexture() . '/' . $this->getCellSize() .
-            '.' . $this->imagesExtension;
-
-        if (file_exists($boardCachedPath)) {
-            return $boardCachedPath;
-        }
+        $boardCachedPath = $this->getCachedFilePath();
 
-        @mkdir($this->cacheDir . '/board/' . $this->getBoardTexture(), 0777, true);
+        try {
+            return new \Imagick($boardCachedPath);
+        } catch (\ImagickException $exception) {
+            @mkdir($this->cacheDir . '/board/' . $this->getBoardTexture(), 0777, true);
 
-        $boardTextureUrl = str_replace('__BOARD_TEXTURE__', $this->getBoardTexture(), $this->boardTextureUrl);
-        $boardTextureUrl = str_replace('__SIZE__', $this->getCellSize(), $boardTextureUrl);
+            $boardTextureUrl = str_replace('__BOARD_TEXTURE__', $this->getBoardTexture(), $this->boardTextureUrl);
+            $boardTextureUrl = str_replace('__SIZE__', $this->getCellSize(), $boardTextureUrl);
 
-        $this->cacheImage($boardTextureUrl, $boardCachedPath);
+            $this->cacheImage($boardTextureUrl, $boardCachedPath);
 
-        return $boardCachedPath;
+            return new \Imagick($boardCachedPath);
```

Comment:
We expect this to work. If it doesn't, exception is caught (and handled) in caller function.